File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.cc
#include <iostream>
using namespace std;

int main() {
    int N, originalN, cycleLength = 0;
    cin >> N;
    originalN = N;

    if (0 <= N && N <= 99) {
        do {
            int tens = N / 10;
            int units = N % 10;
            int sum = tens + units;
            N = units * 10 + sum % 10;
            cycleLength++;
        } while (N != originalN);

        cout << cycleLength << endl;
    } else {
        cout << "입력값은 0부터 99까지의 정수여야 합니다." << endl;
    }

    return 0;
}

File: 백준/Bronze/1110. 더하기 사이클/더하기 사이클.py
import sys

N = int(sys.stdin.readline().rstrip())
originalN = N
cycleLength = 0

while 1:
    tens = N//10
    units = N%10
    N = units*10 + (tens+units)%10
    cycleLength += 1
    if (N == originalN):
        break
print(cycleLength)
#https://velog.io/@dlzagu/%EB%B0%B1%EC%A4%80-1110%EB%B2%88-%ED%8C%8C%EC%9D%B4%EC%8D%AC-%EC%95%8C%EA%B3%A0%EB%A6%AC%EC%A6%98-%EB%AC%B8%EC%A0%9C